utils.py

```diff
@@ -223,14 +228,15 @@ def check_files_exist(dict_yml, path_img, path_label, suffix_dict):
                     fname_label = add_suffix(os.path.join(path_label, subject, ses, contrast, filename), suffix_dict[task])
                     if not os.path.exists(fname_label):
                         missing_files_labels.append(fname_label)
+                        missing_suffixes.add(suffix_dict[task])
     if missing_files:
         logging.warning("The following files are missing: \n{}".format(missing_files))
         logging.warning("\nPlease check that the files listed in the yaml file and the input path are correct.\n")
     if missing_files_labels:
         logging.warning("If you are creating label(s) from scratch, ignore the following message.")
         logging.warning("\nThe following label files are missing: \n{}".format(missing_files_labels))
         logging.warning("\nPlease check that the used suffix '{}' is correct. "
-                        "If not, you can provide custom suffix using '-suffix-files-' flags.\n".format(suffix_dict[task]))
```

Comment:
This was a bug: it always used the `task` value from the last iteration of the loop above, regardless of whether that was the task with `missing_files_labels`, and regardless of whether it's present in `suffix_dict`.

Reply:
Good catch! Thanks for the fix!